server/src/main/java/org/apache/iotdb/db/mpp/operator/meta/TimeSeriesMetaScanOperator.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.mpp.operator.meta;
+
+import org.apache.iotdb.db.exception.metadata.MetadataException;
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.db.mpp.execution.SchemaDriverContext;
+import org.apache.iotdb.db.mpp.operator.OperatorContext;
+import org.apache.iotdb.db.mpp.sql.planner.plan.node.PlanNodeId;
+import org.apache.iotdb.db.qp.physical.sys.ShowTimeSeriesPlan;
+import org.apache.iotdb.db.query.dataset.ShowTimeSeriesResult;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.read.common.block.TsBlock;
+import org.apache.iotdb.tsfile.read.common.block.TsBlockBuilder;
+import org.apache.iotdb.tsfile.utils.Binary;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class TimeSeriesMetaScanOperator extends MetaScanOperator {
+  private String key;
+  private String value;
+  private boolean isContains;
+
+  // if is true, the result will be sorted according to the inserting frequency of the timeseries
+  private boolean orderByHeat;
+
+  private static final TSDataType[] resourceTypes = {
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT,
+    TSDataType.TEXT
+  };
+
+  public TimeSeriesMetaScanOperator(
+      OperatorContext operatorContext,
+      int limit,
+      int offset,
+      PartialPath partialPath,
+      String key,
+      String value,
+      boolean isContains,
+      boolean orderByHeat,
+      boolean isPrefixPath,
+      List<String> columns) {
+    super(operatorContext, limit, offset, partialPath, isPrefixPath, columns);
+    this.isContains = isContains;
+    this.key = key;
+    this.value = value;
+    this.orderByHeat = orderByHeat;
+  }
+
+  public String getKey() {
+    return key;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  public boolean isContains() {
+    return isContains;
+  }
+
+  public boolean isOrderByHeat() {
+    return orderByHeat;
+  }
+
+  @Override
+  protected TsBlock createTsBlock() {
+    TsBlockBuilder builder = new TsBlockBuilder(Arrays.asList(resourceTypes));
+    try {
+      ((SchemaDriverContext) operatorContext.getInstanceContext().getDriverContext())
+          .getSchemaRegion()
+          .showTimeseries(convertToPhysicalPlan(), operatorContext.getInstanceContext())
+          .left
+          .forEach(series -> setColumns(series, builder));
+    } catch (MetadataException e) {
+      throw new RuntimeException(e.getMessage(), e);
+    }
```

Comment:
Should there be a ```RuntimeException```? Do we have any related exception handle framework? @JackieTien97